### PR TITLE
Add rating per user.

### PR DESCRIPTION
**Pull request by ildyria**
## Summary by CodeRabbit

* **New Features**
  * Photo star rating (0–5) with API endpoint, rating widget, keyboard shortcuts, and per-photo ratings.
* **Config**
  * New settings to control rating visibility and whether averages or user ratings are shown per view.
* **Data & Migrations**
  * New rating storage and added statistics fields for rating sum/count/average with migrations and factory defaults.
* **UI**
  * New rating components and overlays; UI updates reflect changes immediately.
* **Docs & Translations**
  * Spec, plan, docs, and translations added for rating UI.
* **Tests**
  * Integration and concurrency tests covering rating behavior and edge cases.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: app/Models/PhotoRating.php [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+	protected $fillable = [
+		'photo_id',
+		'user_id',
+		'rating',
+	];
+
+	protected $casts = [
+		'rating' => 'integer',
+		'user_id' => 'integer',
+	];
```
[the remark this answers is not in the document]
Comment (ildyria) [reply app/Models/PhotoRating.php:51]: Yes that is intended to make sure that we do not get results. when guest.

File: lang/de/gallery.php [hunk 2]
```diff
@@ -181,6 +181,19 @@
...
+            'ratings' => 'ratings',
+            'your_rating' => 'Your rating',
+            'saving' => 'Saving rating...',
+            'removed' => 'Rating removed',
+            'saved' => 'Rating saved',
+            'error' => 'Failed to save rating',
+            'error_unauthorized' => 'You must be logged in to rate photos',
+            'error_forbidden' => 'You do not have permission to rate this photo',
+            'error_not_found' => 'Photo not found',
+        ],
```
[the remark this answers is not in the document]
Comment (ildyria) [reply lang/de/gallery.php:196]: No. As said before do not worry about translations...

File: lang/fa/gallery.php [hunk 3]
```diff
@@ -182,6 +182,19 @@
...
+            'ratings' => 'ratings',
+            'your_rating' => 'Your rating',
+            'saving' => 'Saving rating...',
+            'removed' => 'Rating removed',
+            'saved' => 'Rating saved',
+            'error' => 'Failed to save rating',
+            'error_unauthorized' => 'You must be logged in to rate photos',
+            'error_forbidden' => 'You do not have permission to rate this photo',
+            'error_not_found' => 'Photo not found',
+        ],
```
[the remark this answers is not in the document]
Comment (ildyria) [reply lang/fa/gallery.php:197]: No. As said before do not worry about translations...

File: resources/js/components/icons/StarRow.vue [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+			/>
+			<i
+				:class="{
+					absolute: true,
+					'pi pi-star-fill text-amber-500': index <= Math.floor(props.rating),
+					'text-xs': props.size === 'small',
+					'text-lg': props.size === 'medium',
+					'text-xl': props.size === 'large',
+				}"
+			/>
```
[the remark this answers is not in the document]
Comment (ildyria) [reply resources/js/components/icons/StarRow.vue:39]: Fixed: b809b74909155c1b17474b66aff1c2548327e396

File: resources/js/components/gallery/photoModule/PhotoRatingOverlay.vue [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+		:class="{
+			'absolute bottom-0 w-full sm:w-1/2 left-1/2 -translate-x-1/2 z-20 sm:h-1/8 h-14': true,
+			'lg:hover:opacity-100 transition-opacity duration-500 ease-in-out': isHoverMode && !isTouchDevice(),
+			'opacity-50 lg:opacity-20': isHoverMode && !isTouchDevice() && !isFullTransparency,
+			'opacity-75': isHoverMode && isTouchDevice() && !isFullTransparency,
+			'opacity-0': isHoverMode && isFullTransparency,
+			hidden: is_slideshow_active,
+		}"
+	>
+		<div class="absolute left-1/2 -translate-1/2 bottom-7 flex items-center gap-1 text-shadow">
```
[the remark this answers is not in the document]
Comment (ildyria) [reply resources/js/components/gallery/photoModule/PhotoRatingOverlay.vue]: Fixed: b809b74909155c1b17474b66aff1c2548327e396
